.github/workflows/build_packages.yaml

```diff
@@ -331,25 +342,26 @@ jobs:
             echo "$(cat $package_name.tar.gz.sha256) $package_name.tar.gz" | sha256sum -c || exit 1
           fi
         done
+        fi
+
     - name: build emqx packages
+      working-directory: source
       env:
-        OTP: ${{ matrix.otp }}
         PROFILE: ${{ matrix.profile }}
         ARCH: ${{ matrix.arch }}
         SYSTEM: ${{ matrix.os }}
-      working-directory: source
       run: |
-        ./scripts/buildx.sh \
-          --profile "${PROFILE}" \
-          --pkgtype "tgz" \
-          --arch "${ARCH}" \
-          --builder "ghcr.io/emqx/emqx-builder/5.0-3:${OTP}-${SYSTEM}"
-        ## the pkg build is incremental on the tgz build
-        ./scripts/buildx.sh \
-          --profile "${PROFILE}" \
-          --pkgtype "pkg" \
-          --arch "${ARCH}" \
-          --builder "ghcr.io/emqx/emqx-builder/5.0-3:${OTP}-${SYSTEM}"
+        set -eu
```

Comment:
We’d still need the builders?
To build centos packages on Ubuntu host

Reply:
buildx.sh is a docker cmd wrapper. 

The entire job is already converted to run in the container on {{  matrix.build_machine }} with the desired arch , see L233 and L235, build centos packages on Ubuntu runner will be done and it will work.


There are two reasons to run the job in the container, not on the runner.
- we don't need to maintain the toolchains (rebar3, erlang for rebar3) on the runner, installing erlang took time. Too slow for autoscaling instances and during my test I found there are other missing tools as well.
-  Align with other jobs for the workdir/buildpath so that the *CMake* build-cache will not complain about the path differences.


Now I see we have duplicated code in this task and in `./scripts/buildx.sh`, /scripts/buildx.sh may be kept for developers who want to cross-build on a local x86 machine?

I think the better approach is to update the buildx.sh  instead?

Reply:
use buildx.sh or not, we need to build packages in builder container, but not the runner host
so we have reproducible builds, pined otp (and a lot other dependencies) versions.

Reply:
sorry, now I noticed L233 to L235.